[EDU-2289] fix: review marketplace guides (#421)
aziontech/docs: src/content/docs/en/pages/guides/marketplace/integrations/bot-protection/index.mdx

Using advanced algorithms that analyze the behavior of incoming data, the integration can detect and prevent malicious activities such as credential stuffing, vulnerability scanning, and site scraping.

The system employs *Reputation Intelligence* to establish a profile of each person who visits the site, including location, device type, and browsing patterns. This enables the prompt detection of suspicious behavior and the implementation of preventive measures against potential threats.

---

## Getting the integration

To use Azion **Bot Protection**, access [Real-Time Manager (RTM)](/en/documentation/products/guides/how-to-access-rtm/) and go to **Marketplace**.

On the Marketplace's homepage, select the integration's card.

Once the integration's page opens, click the **Get It Now** button, at the bottom-right corner of the page, and a successful message appears to confirm your integration is installed.

:::tip

> You can search any integration by browsing through the cards or typing a keyword in the search box. 

Reviewer: box ou bar?

> You can search any integration by browsing through the cards or typing a keyword in the search box. 

Author: ```suggestion
You can search any integration by browsing through the cards or typing a keyword in the search bar. 
```

:::

---

## Configuiring the integration

### Setting up the Edge Firewall rule 

To instantiate Azion **Bot Protection**, follow the steps:

1. On the upper-left corner, select **Products menu** > **Edge Firewall** in the **SECURE** section.
2. Click the **Add Rule Set** button.
3. Give an easy-to-remember name to your new rule.
4. Select the domains you want to protect with the function.
5. Enable the **Edge Functions** switch in the **Edge Firewall Modules** section.
6. Click the **Save** button.

Done. Now you've instantiated the rule for your function and have access to edge functions on your edge firewall.

> ### Setting up the function 

Reviewer: ```suggestion
### Setting up the function
```

While still on the **Edge Firewall** page:

1. Select the **Functions** tab.
2. Click the **Add Function** button.
3. Give an easy-to-remember name to your instance.
4. On the dropdown menu, select the **Azion Bot Protection** function.

> This action will load the function, showing a form with the function code and, just above it, two tabs: **Code** and **Arguments**. On the **Arguments** tab, you’ll pass your variables.

Reviewer: é arguments e não args mais?

Author: Eu revise no RTM e apareceu Arguments mesmo

```json
{
  "threshold": 10,
  "action": "deny"
}
```

Even when `threshold` and `action`are the mandatory variables to be defined, you can add and define more variables, according to your needs, as showed in the example below that includes [integration with a SIEM](#integrating-with-a-siem).

```json
{
  "threshold": 10,
  "action": "deny",
  "disabled_rules": [],
  "reputation_network_lists": [2357, 2358, 2358, 2360],
  "log_tag": "log_tag",
  "siem_integration": true,
  "send_data_to_siem_when": "always",
  "siem_connection_args": {
    "host": "example.com",
    "uri": "/inspect/",
    "port": "443",
    "scheme": "https",
    "additional_headers": {
      "X-Test": "a test goes here",
      "X-Another-Test": "another test goes here"
    }
  }
}

> ````

Author: ```suggestion
```
```

> ````

Author: ```suggestion
```
```

Where:

| Variable | Type | Required | Description |
|---|---|---|---|
| `threshold` | Number | Yes | The maximum score that the request can reach before the function takes an action. If it has no value, the function won't take action |
| `action` | String | Yes | The action to be taken by the function whenever the request's score is greater or equals the defined threshold. Possible values: `deny`, `drop`, and `log` (to send the data to the SIEM endpoint, if integrated) |
| `log_tag` | String | No | A tag to identify the function instance which generated the request |
| `log_all_headers` | Boolean | No | Defines whenever or not all the request headers should be sent in the function's log. **Note**: the headers' values are going to be printed with base64 encode |
| `disabled_rules` | Array of numbers | No | The rules to be disabled. If a rule is disabled, it won't be processed nor increment the request score |

> | `reputation_network_lists` | Array of numbers | No | Every Network Lists that should be used to validate the request IP. If the request IP is found in a list, then the request score will be increased. If the request IP is found in multiple lists, the score will be increased in multiple times |

Reviewer 2: ```suggestion
| `reputation_network_lists` | Array of numbers | No | Every network list that should be used to validate the request IP. If the request IP is found in a list, then the request score will be increased. If the request IP is found in multiple lists, the score will be increased in multiple times |
```

| `session_signature_key` | String | No | To sign the signed `az_asm`. If this field has no value or an invalid value, the function will use the default value `azion` |

> | `siem_integration` | Boolean | No | Defines if the request data should be POSTed to an external endpoint. Default value: `false` |

Reviewer: should perform a POST?

> | `siem_integration` | Boolean | No | Defines if the request data should be POSTed to an external endpoint. Default value: `false` |

Author: ```suggestion
| `siem_integration` | Boolean | No | Defines if the request data should perform POST to an external endpoint. Default value: `false` |
```

> | `session_signature_key` | String | No | To sign the signed `az_asm`. If this field has no value or an invalid value, the function will use the default value `azion` |

Author: ```suggestion
| `session_signature_key` | String | No | To sign the `az_asm`. If this field has no value or an invalid value, the function will use the default value `azion` |
```

| `send_data_to_siem_when` | String | No | Defines when the request data should be sent to the external endpoint (if `siem_integration` is enabled). Possible values: `always`, 'deny', `drop`, or `log` (to send the data to the SIEM endpoint, if integrated) |
| `siem_connection_args` | Table | Yes, <br/> with `siem_integration` enabled | The arguments to be used in the POST to the external endpoint |

> | `siem_connection_args.host` | String | Yes,<br/> with `siem_integration` enabled | The host where the data will be POSTed to. If this field is empty, the function will be bypassed |

Reviewer: precisa dos `<br/>`?

Author: fica a coluna muito larga e quebra feio, foi o melhor visual que consegui

| `siem_connection_args.uri` | String | No | The URI where the data will be POSTed to. Default value: `/`  |
| `siem_connection_args.port` | Number | No | The port where the data will be POSTed to. Default value: `443` |
| `siem_connection_args.scheme` | String | No | Defines the scheme to be used in the POST. Default value: `https` |
| `siem_connection_args.additional_headers` | Table | No | Additional headers to be sent in the POST. The function will always send the following headers: `["Accept"] = "application/json"`, `["Content-Type"] = "application/json"` |

When you're done, click the **Save** button.

:::tip
To learn how to get the logs of Azion **Bot Protection**, go to the [Integrating with a SIEM](#integrating-with-a-siem) section.
:::

### Setting up the Rules Engine

To finish, you have to set up a **Rules Engine** to configure the *behavior* and the *criteria* to run the function.

Still in the **Edge Firewall** page:

1. Select the **Rules Engine** tab.
2. Click the **New Rule** button.
3. Give an easy-to-remember name to the rule.
   - You can add a description, but it's an optional step.

> 4. Select a *criteria* to run and catch the domains that you want to run the integration on. 

Reviewer: ```suggestion
4. Select a *criteria* to run and catch the domains that you want to run the integration on.
```

    - Use this rule: `if Request URI does not match "\.(png|jpg|css|js|jpeg|gif|ico|ttf|svg|woff|woff2|ashx|asmx|svc|swf|otf|eot)(\?.*)?$"`
        - This rule is necessary to exclude all static data on your application.
5. You have to create another criteria for this integration to work: `if Request URI does not match /.well-know/`
    - This rule is necessary to create a list of allowed IPs that doesn't impact automation or scripts to WEB API.

> 6. Below, select a *behavior* to the *criteria*. In this case, it’ll be **Run Function**.
>   - Select the adequate function according to the name you gave it in the instantiate step.
> 7. Click the **Save** button.

Author: ```suggestion
6. Below, select a *behavior* to the *criteria*. In this case, it’ll be **Run Function**.
7. Select the adequate function according to the name you gave it in the instantiate step.
8. Click the **Save** button.
```

}
```

> ### Log Example

Reviewer: ```suggestion
### Log example
```

---

## Important

> Azion will provide you with easy-to-go configurations, that should be enough for most of the cases. If you need a more detailed configuration, such as custom rules, you can edit the`JSON` file for the integration.

Reviewer: ```suggestion
Azion will provide you with easy-to-go configurations, that should be enough for most of the cases. If you need a more detailed configuration, such as custom rules, you can edit the `JSON` file for the integration.
```
